modules/installation/pages/upgrade.adoc

```diff
@@ -85,6 +85,128 @@ $ gadmin restart all -y
+
+After the binaries and configuration files are installed, user will be prompted with the following message:
+
+[console]
+----
+"Do you want to switch platform to the new version now (it can be delayed to a later time): y/N"
+----
+
+. n/N option
+
+Selecting n/N will guide the user through the manual steps to switch to version 4.2.1, assuming the user is upgrading from 4.2.0.
+
+[console]
+----
+To switch platform to new version, please run 'gadmin upgrade 4.2.1'
+----
+
+. y/Y Option
+
+If users selects `y/Y' the installer will automatically switch TigerGraph to the latest maintenance version. 
+If any errors occur, the upgrade process will terminate immediately. Follow the instructions in the prompts to manually continue the upgrade or revert to the original version.
+
+. Option `-n`
+
+If the user wants to skip manual input during the upgrade, they can use the `-n` flag, which automatically switches to the new version without requiring any interaction.
+
+[source, gsql]
+----
+$ ./install.sh -Ur -n
+----
+
+.Example:
+[source, gsql]
+----
+./install.sh -Ur -n
+
+   _______                 ______                 __
+  /_  __(_)___ ____  _____/ ____/________ _____  / /_
+   / / / / __ `/ _ \/ ___/ / __/ ___/ __ `/ __ \/ __ \
+  / / / / /_/ /  __/ /  / /_/ / /  / /_/ / /_/ / / / /
+ /_/ /_/\__, /\___/_/   \____/_/   \__,_/ .___/_/ /_/
+       /____/                          /_/
+
+
+Welcome to the TigerGraph platform installer!
+
+[PROGRESS]: 08:10:44 Upgrade TigerGraph platform from existing one ... 
+[NOTE    ]: Obtained version of exist platform: 4.2.0 [OK] 
+[NOTE    ]: Obtained version of new platform: 4.2.1 [OK] 
+----
+
+. Handling Downtime
+
+During the rolling upgrade, expect a temporary service disruption. If the upgrade fails or the connection is lost, user can *resume* the upgrade process by running:
+
+[source.wrap,bash]
+----
+$ gadmin upgrade --resume
+----
+
+. Revert to Previous Version
+
+To revert to your original version, follow these steps:
+
+[source.wrap,bash]
+----
+$ gadmin config set System.AppRoot /home/tigergraph/tigergraph/app/4.2.0
+$ gadmin config apply -y
+$ gadmin restart all -y
+----
+
+If the upgrade fails for any reason, the error prompt will guide you on whether to continue manually or revert to the previous version. 
+Use the following command if you need to manually finish the upgrade or revert:
+
+*Manual Upgrade*: Execute the following commands to restart services in sequence:
+
+[source.wrap,bash]
+----
+gadmin restart GUI#1 ADMIN#1 NGINX#1 EXE_1 IFM#1 KAFKASTRM-LL_1 RESTPP#1 DICT#1 KAFKACONN#1 GSE_1#1 GPE_1#1 CTRL#1 DICT#2 KAFKACONN#2 RESTPP#2 IFM#2 GUI#2 EXE_2 ADMIN#2 KAFKASTRM-LL_2 CTRL#2 GSE_1#2 NGINX#2 GPE_1#2 NGINX#3 GPE_1#3 GUI#3 KAFKASTRM-LL_3 DICT#3 KAFKACONN#3 RESTPP#3 EXE_3 ADMIN#3 CTRL#3 GSE_1#3 IFM#3 --no-dep -y
+----
+
+*Long-Running Operations*
+
+Before starting the upgrade process, ensure that long-running operations are allowed by running the following command:
```

Comment:
If this step is for "before starting the upgrade", shouldn't this be at the beginning, as a Prerequisite?